client/src/pages/donate-other.js

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable max-len */
+import React, { Component, Fragment } from 'react';
+import Helmet from 'react-helmet';
+import ReactGA from 'react-ga';
+import { Grid, Col, Row } from '@freecodecamp/react-bootstrap';
+
+import Spacer from '../components/helpers/Spacer';
+
+const payPalPayments = [
+  { eventLabel: 'paypal',
+    eventValue: 5,
+    defaultValueHash: 'KTAXU8B4MYAT8',
+    defaultValue: 'Donate $5 each month'
+  },
+  { eventLabel: 'paypal',
+    eventValue: 10,
+    defaultValueHash: 'BYEBQEHS5LHNC',
+    defaultValue: 'Donate $10 each month'
+  },
+  { eventLabel: 'paypal',
+    eventValue: 35,
+    defaultValueHash: '57ZPMKJ8G893Y',
+    defaultValue: 'Donate $35 each month'
+  },
+  { eventLabel: 'paypal',
+    eventValue: 50,
+    defaultValueHash: '2ZVYTHK6Q7AFW',
+    defaultValue: 'Donate $50 each month'
+  },
+  { eventLabel: 'paypal',
+    eventValue: 100,
+    defaultValueHash: 'C7PUT3LMJHKK2',
+    defaultValue: 'Donate $100 each month'
+  },
+  { eventLabel: 'paypal',
+    eventValue: 250,
+    defaultValueHash: '69JGTY4DHSTEN',
+    defaultValue: 'Donate $250 each month'
+  },
+  { eventLabel: 'paypal one time donation',
+    eventValue: 0,
+    defaultValueHash: 'B256JC6ZCWD3J',
+    defaultValue: 'Make a one-time donation'
+  }
+];
+
+class IndexPage extends Component {
+    constructor(props, context) {
+    super(props, context);
+  }
+
+  formbuilder(eventLabel, eventValue, defaultValueHash, defaultValue) {
+    return (
+      <div onClick={()=> ReactGA.event({category: 'Donation', action: 'paypal', label: eventValue.toString()})}>
+        <form
+          action='//www.paypal.com/cgi-bin/webscr'
+          method='post'
+          onSubmit={`ga('send',  { hitType: 'event', eventCategory: 'donation', eventAction: 'click', eventLabel: ${eventLabel} 'paypal', eventValue: ${eventValue} } );`}
+          target='_blank'
+          >
+          <input defaultValue='_s-xclick' name='cmd' type='hidden' />{' '}
+          <input
+            defaultValue={defaultValueHash}
+            name='hosted_button_id'
+            type='hidden'
+          />{' '}
+          <input
+            className='btn btn-cta signup-btn btn-block'
+            defaultValue={defaultValue}
+            name='submit'
+            type='submit'
+          />
+        </form>
+      </div>
+    );
+  }
+
+  formfilter = (array, frequency) => {
+    if (frequency === 'monthly') {
+      return array
+        .filter(item => item.eventValue > 0)
+        .map((item) => {
+          return this.formbuilder(item.eventLabel, item.eventValue, item.defaultValueHash, item.defaultValue);
```

Comment:
It would be cleaner to just pass the entire `item` to the `formBuilder`, you can then extract what is needed from `item`. This method doesn't look like it needs to know the details of each item.

Imagine we needed to add/remove a property from `item`, instead of working just where `item` is consumed (in this case, to render the form), we have to travel along with it touching code that shouldn't need to change.